rewriter/DefDelegator.cc

```diff
@@ -0,0 +1,74 @@
+#include "rewriter/DefDelegator.h"
+#include "ast/Helpers.h"
+#include "core/GlobalState.h"
+#include "rewriter/Util.h"
+#include <optional>
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+vector<ast::TreePtr> DefDelegator::run(core::MutableContext ctx, const ast::Send *send) {
+    vector<ast::TreePtr> empty;
+    auto loc = send->loc;
+
+    if (send->fun != core::Names::defDelegator()) {
+        return empty;
+    }
+
+    // This method takes 2..3 positional arguments and no keyword args:
+    // `def_delegator(accessor, method, ali = method)`
+    if (!(send->numPosArgs == 2 || send->numPosArgs == 3)) {
+        return empty;
+    }
+
+    if (send->hasKwArgs()) {
+        return empty;
+    }
+
+    auto *accessor = ast::cast_tree<ast::Literal>(send->args[0]);
+    auto *method = ast::cast_tree<ast::Literal>(send->args[1]);
```

Comment:
Could you move the declaration of method to right above the conditional where it's tested?